pkg/kv/dist_sender.go

```diff
@@ -1235,115 +1214,29 @@ func (ds *DistSender) sendToReplicas(
 					propagateError = true
 				}
 
-				log.ErrEventf(ctx, "application error: %s", call.Reply.Error)
-
 				if propagateError {
-					if pending > 0 {
-						// If there are still pending RPCs, try to wait them out.
-						//
-						// Note that ambiguous result errors can arrive from these
-						// in-flight RPCs, so if we've sent any, we better wait
-						// for them, even if we're not going to generate ambiguous
-						// results from them ourselves.
-						timer := time.NewTimer(ds.pendingRPCTimeout)
-						defer timer.Stop()
-					pendingLoop:
-						for pending > 0 {
-							select {
-							case pendingCall := <-done:
-								pending--
-								if err := pendingCall.Err; err != nil {
-									if grpc.Code(err) != codes.Unavailable {
-										ambiguousError = err
-									}
-								} else if pendingCall.Reply.Error == nil {
-									return pendingCall.Reply, nil
-								}
-							case <-timer.C:
-								break pendingLoop
-							}
-						}
-					}
-
 					// The error received is likely not specific to this
 					// replica, so we should return it instead of trying other
-					// replicas. However, if we're trying to commit a
-					// transaction and there are still other RPCs outstanding
-					// or an ambiguous RPC error was already received, we must
-					// return an ambiguous commit error instead of the
-					// returned error.
-					if haveCommit {
-						if pending > 0 || ambiguousError != nil {
-							log.ErrEventf(ctx, "returning ambiguous result (error=%v pending=%d)", ambiguousError, pending)
-							return nil, roachpb.NewAmbiguousResultError(
-								fmt.Sprintf("error=%v pending=%d", ambiguousError, pending))
-						}
-					}
+					// replicas.
 					return call.Reply, nil
 				}
 
-				// Extract the detail so it can be included in the error
-				// message if this is our last replica.
-				//
+				log.ErrEventf(ctx, "application error: %s", call.Reply.Error)
+			}
+
+			if transport.IsExhausted() {
```

Comment:
Now that you've removed the `SendNextTimeout()`, nothing uses the `deadline` being set in the `grpcTransport`. This means that in the case where node2 and node3 believe node1 (AWOL) is still the leaseholder, node1 will keep getting moved to the front of the transport, will fail fast, then we'll try node2 and node3 in rapid succession, each putting node1 back into the front position.

The end result is that because you're no longer employing deadline to trigger a timer, this code will busy loop, sending RPCs to node2 and node3 until a Raft election and leaseholder transfer to node2 or node3.